zerver/tests/test_message_fetch.py

```diff
@@ -2671,6 +2675,92 @@ def _update_tsvector_index(self) -> None:
             """
             )
 
+    def test_get_visible_messages_with_narrow_with(self) -> None:
+        hamlet = self.example_user("hamlet")
+        cordelia = self.example_user("cordelia")
+        othello = self.example_user("othello")
+        self.login("hamlet")
+
+        # Test narrow with accessible stream message operand.
+        self.subscribe(self.example_user("hamlet"), "Scotland")
+
+        message_ids = [
+            self.send_stream_message(self.example_user("iago"), "Scotland") for i in range(5)
+        ]
+
+        narrow = [
+            dict(operator="channel", operand="Scotland"),
+            dict(operator="topic", operand="test"),
+            dict(operator="with", operand=f"{message_ids[0]}"),
+        ]
+        self.message_visibility_test(narrow, message_ids, 2)
```

Comment:
I'm not sure why you're using the `message_visibility_test` helper; that was written for 277d463d23f289b66da0df33661e234d33469d82 to help with tests that want to verify that we handle `first_visible_message_id` correctly, but it does a bunch of duplicated work, like fetching the same query twice, that isn't relevant for this operator at all.

Reply:
updated the tests.